back/commons/helpers/api_helper.go

```diff
@@ -0,0 +1,144 @@
+package helpers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+)
+
+type ApiError struct {
+	Error   bool   `json:"error"`
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+func extractFieldNameFromError(errString string, obj any) (string, *string) {
+	if strings.Contains(errString, "is not set") {
+		errParts := strings.Split(errString, " ")
+		fieldName := strings.Trim(errParts[1], "'")
+
+		s := reflect.ValueOf(obj).Elem()
+		for i := 0; i < s.NumField(); i++ {
+			typeField := s.Type().Field(i)
+			jsonTag := typeField.Tag.Get("json")
+			if jsonTag == fieldName {
+				return fieldName, &fieldName
+			}
+		}
+	}
+
+	return "message", nil
+}
```

Comment:
on évite les cascades conditionnelles.
vérifie que ça ne contient pas "is not set", si cette condition est true alors return "message" (pourquoi message, jsp 😂)

Et ensuite tu peux gérer tranquillement comme tu l'as fait ici tout le reste